ddtrace/contrib/internal/openai/utils.py

```diff
@@ -162,6 +162,77 @@ async def _extract_token_chunk(self, chunk):
             return
 
 
+class TracedOpenAIResponseStream(BaseTracedOpenAIStream):
+    def __enter__(self):
+        self.__wrapped__.__enter__()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.__wrapped__.__exit__(exc_type, exc_val, exc_tb)
+
+    def __iter__(self):
+        exception_raised = False
+        try:
+            for chunk in self.__wrapped__:
+                if chunk.type == "response.completed":
+                    handle_response_tools(chunk, self._dd_span)
```

Comment:
it looks like `chunk` has the response object on it like `chunk.response` - i would try that out here! and below for `chunk.usage` -- try `chunk.response.usage`